src/components/Inventory/index.tsx

```diff
@@ -146,11 +163,65 @@ export default function Inventory(props: InventoryProps): JSX.Element {
       facilityIds: filters.facilityIds,
       searchSortOrder,
     });
-    const updatedResult = apiSearchResults?.map((result) => {
-      const resultTyped = result as InventoryResult;
-      const facilityInventoriesNames = resultTyped.facilityInventories.map((nursery) => nursery.facility_name);
-      return { ...result, facilityInventories: facilityInventoriesNames.join('\r') };
-    });
+
+    let updatedResult: InventoryResultWithFacilityNames[] | undefined = [];
+    if (filters.facilityIds && filters.facilityIds.length) {
+      const nextResults: InventoryResultWithFacilityNames[] = [];
+      apiSearchResults?.reduce((acc, result) => {
+        const resultTyped = result as FacilityInventoryResult;
+        const indexFound = acc.findIndex((res) => res.species_id === resultTyped.species_id);
+
+        if (indexFound !== undefined && indexFound !== -1) {
+          const existingSpecies = acc[indexFound];
+          acc[indexFound] = {
+            ...existingSpecies,
+            germinatingQuantity: (
+              Number(existingSpecies.germinatingQuantity) + Number(resultTyped['germinatingQuantity(raw)'])
+            ).toString(),
+            notReadyQuantity: (
+              Number(existingSpecies.notReadyQuantity) + Number(resultTyped['notReadyQuantity(raw)'])
+            ).toString(),
+            readyQuantity: (
+              Number(existingSpecies.readyQuantity) + Number(resultTyped['readyQuantity(raw)'])
+            ).toString(),
+            totalQuantity: (
+              Number(existingSpecies.totalQuantity) + Number(resultTyped['totalQuantity(raw)'])
+            ).toString(),
+            facilityInventories: `${existingSpecies.facilityInventories}, ${resultTyped.facility_name}`,
+          };
+        } else {
+          const transformedResult: InventoryResultWithFacilityNames = {
+            species_id: resultTyped.species_id,
+            species_scientificName: resultTyped.species_scientificName,
+            germinatingQuantity: resultTyped['germinatingQuantity(raw)'],
+            notReadyQuantity: resultTyped['notReadyQuantity(raw)'],
+            readyQuantity: resultTyped['readyQuantity(raw)'],
+            totalQuantity: resultTyped['totalQuantity(raw)'],
+            facilityInventories: resultTyped.facility_name,
+          };
+
+          acc.push(transformedResult);
+        }
+        return acc;
+      }, nextResults);
```

Comment:
Thanks for the changes Constanza. Any chance you can revisit the reduce?
Ideally reduce pattern is 
const xyz =  arr.reduce(..., initialValue);

What you have here may still work but is not ideal, the acc and initial value are references to the original constant in your case, not passed around by value.

Reply:
Thanks, I was replacing the for each with the reduce and didn't realize I was doing that!